### PR TITLE
 Fixes #1768: Integrate messageComposer with redux store  ( Fully functional )

### LINKED ISSUE #1768: Integrate MessageComposer.react.js with redux store
**Actual Behaviour**

Message composer is using the flux store for now.

**Expected Behaviour**

It should be integrated with the new redux store.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AakashMallik**
Fixes #1768

Changes: The changes I have made are as follows:
- [x] Removed all redundant variables. This includes many state variables that were used for no good reason.
- [x] Simplified structure for many functions.
- [x] Fixed a bug where the Mic Icon / Send Icon would not appear until user sent at least one message.

Here are few reasons for why I have made some drastic changes.
* I have created a new util file **messageFilter** that exports a function that is used to seave all the user messages from the list of messages currently available in store.
* As the messages are received via server call so there is a delay, thus I have introduced **componentsWillRecieve** props which when triggered for the first time by the incomming new messages from the server will check the **this.userMessageHistory** array's length. If this array is empty, we fill it using our util filter using the freshly recieved props. ( It is a key improvement that I have made that will ensure this performance heavy **filtering** task will never be done more than once. ). Now from here on, every time the user sends a message, it is simply appended to **this.userMessageHistory**; which outperforms the previous implementation. 
* I have introduced **componetWillMount** to check whether or not the browser supports speechToText feature, this will save us one render cycle.
* array indexing has been done from 0, and if the user is typing a text, the index is changed to -1 giving a more logical vibe.
* I have reused the variable **isListening** in place of previously redundant variables. 
* I have renamed several variables to make them more developer friendly. 

Note: I have done multiple tests to make sure everything checks out, hopefully, I have not missed anything.

Demo Link: https://pr-1849-fossasia-susi-web-chat.surge.sh

### CODE REVIEW

**Comment by AakashMallik** [comment]
@akshatnitd @anshumanv @isuruAb  Sir, this is not the finalized PR so I haven't written neat commit messages. There are also some minor changes like the **snackbar** thing that I would be able to do when this PR   https://github.com/fossasia/chat.susi.ai/pull/1824 gets merged and a few cleanup that I have left out. 
If you feel everything is working fine, give me a green light and I will finalize it. 

Also, the breaking changes will be resolved in https://github.com/fossasia/chat.susi.ai/pull/1849  and https://github.com/fossasia/chat.susi.ai/issues/1712. I have renamed them so I'll be integrating them in the same issue. 

Thank you for your time.

**Comment by AakashMallik** [comment]
@pranav1698 It would be really hard for me to do that. We are doing a component by component refactor/redux integration as you might have noticed. Plus breaking this into multiple PRs would actually result in PRs that are hard to test as this PR itself if atomic in nature. That is why I have left that explanatory comment to make the review process as easy as possible. :sweat_smile: I hope you understand. Sorry.

**Comment by AakashMallik** [comment]
@pranav1698 I am really grateful. Thank you for your time. :innocent:

**Comment by AakashMallik** [comment]
@akshatnitd @anshumanv  I have finalized this PR. It is good to go.

File: src/components/ChatApp/MessageComposer.react.js [hunk 1]
```diff
@@ -77,322 +75,275 @@ class MessageComposer extends Component {
...
-      this.Button = <SendIcon />;
-      this.speechRecog = false;
+      this.speechToTextAvailable = false;
     }
+  }
 
+  componentDidMount() {
+    const { speechOutputAlways, enterAsSend } = this.props;
+    this.userMessageHistory = [];
     let testSkill = $.urlParam('testExample');
```
Comment (AakashMallik) [inline src/components/ChatApp/MessageComposer.react.js]: The $ dependency has been handled by https://github.com/fossasia/chat.susi.ai/pull/1854 submitted by @fragm3  so I have not removed it.

**Comment by AakashMallik** [comment]
@akshatnitd Sir, I'll do the conflict resolution after this gets merged https://github.com/fossasia/chat.susi.ai/pull/1857 else I'll have to resolve conflicts twice. Thank you for your patience.

**Comment by AakashMallik** [comment]
A few updates on this PR:
- I have switched over to **componentDidUpdate** from **componentWIllRecieveProps** to reduce an extra variable that was needed to store the value of the old prop.
- I have fixed a small bug in the MessageSection where the downArrow was visible even when no text was there. 

Rest of the code is more or less the same. 
I have tried most of the possible cases for the MessageComposer, I case I missed something, do let me know. @akshatnitd

File: src/components/ChatApp/MessageComposer.react.js [hunk 2]
```diff
@@ -66,303 +65,278 @@ class MessageComposer extends Component {
...
+      speechRecognitionTextcolor: '#000',
     };
-    this.flag = 1;
+    this.userMessageHistory = [];
   }
 
-  componentDidMount() {
-    const micInputSetting = UserPreferencesStore.getMicInput();
-    if (micInputSetting) {
+  componentWillMount() {
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageComposer.react.js]: cWM lifecycle is deprecated in React latest versions. Please write this block in cDM or constructor as per requirement.

File: src/components/ChatApp/MessageComposer.react.js [hunk 3]
```diff
@@ -19,8 +19,7 @@ import {
 import { urlParam } from '../../utils/helperFunctions';
 import * as apis from '../../apis';
 import VoiceRecognition from './VoiceRecognition';
-import UserPreferencesStore from '../../stores/UserPreferencesStore';
-import MessageStore from '../../stores/MessageStore';
+import { getAllUserMessages } from '../../utils/messageFilter';
```
Comment (anshumanv) [inline src/components/ChatApp/MessageComposer.react.js]: Best. 🎉

Comment (AakashMallik) [reply src/components/ChatApp/MessageComposer.react.js]: Thank you @anshumanv  sir. :innocent:

File: src/components/ChatApp/MessageComposer.react.js (hunk 2 above)
Comment (AakashMallik) [reply src/components/ChatApp/MessageComposer.react.js]: Didn't know that !! I have moved the check in the constructor now. @akshatnitd

**Pull request merged**